hoomd/mpcd/ReverseNonequilibriumShearFlow.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file ReverseNonequilibriumShearFlow.h
+ * \brief Declaration of Reverse nonequilibrium shear flow
+ */
+
+#ifndef MPCD_REVERSE_NONEQUILIBRIUM_SHEAR_FLOW_H_
+#define MPCD_REVERSE_NONEQUILIBRIUM_SHEAR_FLOW_H_
+
+#ifdef NVCC
+#error This header cannot be compiled by nvcc
+#endif
+
+#include "hoomd/ParticleGroup.h"
+#include "hoomd/Updater.h"
+#include <pybind11/pybind11.h>
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+//! Reverse nonequilibrium shear flow updater
+/*!
+ * ReverseNonequilibriumShearFlow implements reverse pertubation flow as
+ * described in Mueller-Plathe,F. Physical Review E 59.5 (1999): 4894.
+ *
+ * A flow is induced by swapping velocities in x direction based on particle position in
+ * z-direction.
+ */
+class PYBIND11_EXPORT ReverseNonequilibriumShearFlow : public Updater
+    {
+    public:
+    //! Constructor
+    ReverseNonequilibriumShearFlow(std::shared_ptr<SystemDefinition> sysdef,
+                                   std::shared_ptr<Trigger> trigger,
+                                   unsigned int num_swap,
+                                   Scalar slab_width,
+                                   Scalar slab_distance,
+                                   Scalar p_target);
+
+    //! Destructor
+    virtual ~ReverseNonequilibriumShearFlow();
+
+    //! Apply velocity swaps
+    virtual void update(uint64_t timestep);
+
+    //! Set the maximum number of swapped pairs
+    void setNswap(unsigned int new_num_swap);
+
+    //! Set the slab width
+    void setSlabWidth(Scalar slab_width);
+
+    //! Set the slab width
+    void setSlabDistance(Scalar slab_distance);
+
+    //! Set the target velocity
+    void setTargetMomentum(Scalar p_target);
+
+    //! Get max number of swaps
+    Scalar getNswap() const
+        {
+        return m_num_swap;
+        }
+
+    //! Get slab thickness
+    Scalar getSlabWidth() const
+        {
+        return m_slab_width;
+        }
+
+    //! Get slab thickness
+    Scalar getSlabDistance() const
+        {
+        return m_slab_distance;
+        }
+
+    //! Get target momentum
+    Scalar getTargetMomentum() const
+        {
+        return m_p_target;
+        }
+
+    //! Computes the requested log value and returns it
+    virtual Scalar getLogValue(const std::string& quantity, uint64_t timestep);
+
+    //! Returns a list of log quantities this compute calculates
+    virtual std::vector<std::string> getProvidedLogQuantities();
```

Comment:
Remove these and replace them with something like `Scalar getTotalSwappedMomentum() const;` that will return the total swapped momentum. Then, expose this getter as a readonly Python property.